src/ui/Templates/TemplateToHtml.ts

```diff
@@ -0,0 +1,71 @@
+import { IQueryResults } from '../../rest/QueryResults';
+import { IQueryResult } from '../../rest/QueryResult';
+import { Assert, QueryUtils, Component, Initialization, QueryStateModel } from '../../Core';
+import { IInitResult } from '../Base/Initialization';
+import { Template } from './Template';
+import { SearchInterface } from '../../ui/SearchInterface/SearchInterface';
+import { ResultList } from '../ResultList/ResultList';
+import { ValidLayout } from '../ResultLayoutSelector/ValidLayout';
+
+export interface ITemplateToHtml {
+  resultTemplate: Template;
+  queryStateModel: QueryStateModel;
+  searchInterface: SearchInterface;
+}
+
+export class TemplateToHtml {
+  constructor(public args: ITemplateToHtml) {}
+
+  /**
+   * Builds and returns an array of HTMLElement with the given result set.
+   * @param results the result set to build an array of HTMLElement from.
+   */
+  public async buildResults(
+    results: IQueryResults,
+    layout: ValidLayout,
+    currentlyDisplayedResults: IQueryResult[]
+  ): Promise<HTMLElement[]> {
+    const resultsPromises = results.results.map((result: IQueryResult) => {
+      return this.buildResult(result, layout, currentlyDisplayedResults);
+    });
+
+    const resultElement = await Promise.all(resultsPromises);
```

Comment:
Unfortunately, `Promise.all `does not work in IE11, and I'd prefer if we had a way to not have to add polyfill.

The reason is because it's not as simple as "simply adding a polyfill". 

For example, websites might already provide their own "Promise library" in JS (and override the native one), and then the polyfill might not be added on the proper object (eg: it adds on the native IE11 Promise object, but not the one implemented in JS through a library).

We've had problem with this in the past (and still happen from time to time), so the less of these kind of things we have, the better.

Reply:
ok, I'll revert this change then.